test/utility/parse_opts.c

```diff
@@ -26,19 +26,82 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 #include "test_util.h"
+#define DIR_STORE "dir_store"
 
 extern char *__wt_optarg; /* argument associated with option */
 
+/*
+ * parse_tiered_opts --
+ *     Parse command line options for the tiered storage configurations.
+ */
+static inline void
+parse_tiered_opts(int argc, char *const *argv, TEST_OPTS *opts)
+{
+    int index;
+    int number_of_tiered_options;
+
+    number_of_tiered_options = 0;
+    opts->tiered_storage = false;
+
+    for (int i = 1; i < argc; i++) {
+        /* Tiered storage command line options starts with -P. */
+        if (strstr(argv[i], "-P")) {
+            /* Many more options to come here. */
+            if (argv[i][2] == 'o') {
+                if (argv[i + 1] == NULL)
+                    testutil_die(
+                      EINVAL, "%s option requires an argument %s", opts->progname, argv[i]);
+                number_of_tiered_options += 2;
+            } else if (argv[i][2] == 'T') {
+                /* This parsing is different because -PT does not accept any arguments. */
+                ++number_of_tiered_options;
+                opts->tiered_storage = true;
+            }
+        }
+    }
+
+    /* Return from here if tiered arguments are not passed. */
+    if (!opts->tiered_storage)
+        return;
+
+    opts->nargc = argc - number_of_tiered_options;
+
+    /* Allocate the memory for the new argv without tiered options. */
+    opts->nargv = dmalloc((size_t)opts->nargc);
+    for (int i = 0; i < opts->nargc; ++i) {
+        opts->nargv[i] = dmalloc(1024);
+    }
+
+    opts->tiered_storage = false;
+    index = 0;
+
+    /* Copy other command line arguments except tiered. */
+    for (int i = 0; i < argc; i++) {
+        if (strcmp(argv[i], "-Po") == 0) {
+            opts->tiered_storage_source = dstrdup(argv[i + 1]);
+            /* Move the index because this option has an argument. */
+            i++;
+        } else
+            opts->nargv[index++] = dstrdup(argv[i]);
```

Comment:
Don't you need to check against `-PT` here also?

Reply:
Fixed